pkg/virt-controller/network/pod_annotations.go

```diff
@@ -31,19 +31,35 @@ import (
 )
 
 func GeneratePodAnnotations(networks []virtv1.Network, interfaces []virtv1.Interface, multusStatusAnnotation string, bindingPlugins map[string]virtv1.InterfaceBindingPlugin) map[string]string {
-	newAnnotations := map[string]string{}
+	var err error
+
+	var sriovIfacePciMap map[string]*networkv1.DeviceInfo
 	if vmispec.SRIOVInterfaceExist(interfaces) {
-		networkPCIMapAnnotationValue := deviceinfo.CreateNetworkPCIAnnotationValue(
-			networks, interfaces, multusStatusAnnotation,
-		)
-		newAnnotations[deviceinfo.NetworkPCIMapAnnot] = networkPCIMapAnnotationValue
+		sriovIfacePciMap, err = deviceinfo.MapNetworkNameToDeviceInfo(networks, multusStatusAnnotation, interfaces)
+		if err != nil {
+			log.Log.Warningf("failed to create network-device-info-map for SR-IOV interfaces: %v", err)
+		}
 	}
+
+	var pluginIfacePciMap map[string]*networkv1.DeviceInfo
 	if vmispec.BindingPluginNetworkWithDeviceInfoExist(interfaces, bindingPlugins) {
-		networkDeviceInfoMap, err := deviceinfo.MapBindingPluginNetworkNameToDeviceInfo(networks, interfaces, multusStatusAnnotation, bindingPlugins)
+		pluginIfacePciMap, err = deviceinfo.MapBindingPluginNetworkNameToDeviceInfo(networks, interfaces, multusStatusAnnotation, bindingPlugins)
 		if err != nil {
 			log.Log.Warningf("failed to create network-device-info-map: %v", err)
-			networkDeviceInfoMap = map[string]*networkv1.DeviceInfo{}
 		}
+	}
+
+	networkDeviceInfoMap := map[string]*networkv1.DeviceInfo{}
+	for k, v := range pluginIfacePciMap {
+		networkDeviceInfoMap[k] = v
+	}
+	for k, v := range sriovIfacePciMap {
+		networkDeviceInfoMap[k] = v
+	}
```

Comment:
Please extract it to a helper that merges maps.

Reply:
It seems that since Go 1.21 maps package is graduated and now part of the standard libs https://pkg.go.dev/maps@go1.21.0#Copy , we can use maps.Copy instead :slightly_smiling_face:

Reply:
Nice.
Still, please extract it to a helper.